src/stages/runTest.ts

```diff
@@ -1,16 +1,36 @@
 import { exec } from '@actions/exec';
 
+import { getErrorMessage, toErrorWithMessage } from '../utils/getErrorMessage';
 import { getTestCommand } from '../utils/getTestCommand';
 
+const THRESHOLD_NOT_MET_REGEX =
+    /Jest:.*coverage threshold for.*not met:.*%\S/gm;
+
 export const runTest = async (
     testCommand: string,
     workingDirectory?: string
 ) => {
-    await exec(
-        await getTestCommand(testCommand, 'report.json', workingDirectory),
-        [],
-        {
-            cwd: workingDirectory,
+    try {
+        await exec(
+            await getTestCommand(testCommand, 'report.json', workingDirectory),
+            [],
+            {
+                cwd: workingDirectory,
+            }
+        );
+    } catch (error) {
+        const errMessage = getErrorMessage(error) ?? '';
+
+        if (THRESHOLD_NOT_MET_REGEX.test(errMessage)) {
+            const newErr = toErrorWithMessage(error);
+            const [newMessage] =
+                errMessage.match(THRESHOLD_NOT_MET_REGEX) ?? [];
+
+            newErr.message = newMessage;
+
+            throw newErr;
         }
-    );
+
+        throw error;
```

Comment:
The problem is, that `exec` will throw error with message `The process 'npx' failed with exit code 1` ([i've built your changes to test them in separate repo](https://github.com/ArtiomTr/jest-coverage-report-action-test/pull/16)). So you cannot extract the reason of failure, by this message.

Alternatively, you can collect `stderr` output from exec command, and then parse it, when receiving exception. For instance:
```ts
const errorOutput = '';
await exec(
            await getTestCommand(testCommand, 'report.json', workingDirectory),
            [],
            {
                cwd: workingDirectory,,
                stderr: (data: Buffer) => {
                  errorOutput += data.toString();
                }
            }
        );
```

Reply:
Also, I don't really know how to deal with another problem. The threshold check results already have output:
![image](https://user-images.githubusercontent.com/44021713/160254490-b23786c9-9664-4dba-b621-7f4a4369d86b.png)
So returning exception with detailed output will duplicate information in the report. This could be avoided by silently ignoring exception. However, this is bad solution, as this may lead to unpredictable behavior. I do not have a clear understanding of how to fix this, so you can implement as you see fit.

Reply:
Interesting @ArtiomTr - I am not seeing that error message on our test runs: 
<img width="648" alt="image" src="https://user-images.githubusercontent.com/8508556/160693665-fc4d0942-c32c-402f-8255-1c245c2188a3.png">


The resulting failure from the above image is that the `jest threshold` was not met for a specific directory. I'm guessing the error messaging about `statements` and `functions` are the `global threshold`s? While useful - we've targeted directories themselves for coverage thresholds so that we can make sure that as we increase our coverage we don't lose coverage in specific areas as well. 

Looking over the code a bit more - I'm wondering if we shouldn't just utilize the above method to capture `stdout` that you outlined and then use `dataCollector` to push those errors onto the `dataCollector.errors` array. From there `formatErrors` looks like it can handle everything. 

Let me take a quick look and see if I can't get something running.

Reply:
@therynamo, I guess the threshold information reporting not works for the directory thresholds. But, that is a separate issue that should be addressed in another PR. To confirm, that this is a bug, not related to this issue, you could try to modify your threshold configuration, by specifying glob patterns for **files**, instead of directories. For example:

```diff
coverageThreshold: {
-   dirname: {
+   "dirname/**/*.js": {
        // ...
    }
  },
```

About using `formatErrors` function as a mechanism for reporting threshold errors - I would like to separate the threshold messages from the "errors", because I think these have different meanings:
1. Thresholds not met - action is doing everything correctly, there is an issue with tests.
2. Errors - some bug / unexpected behavior / bad configuration.

So to make messages more clear, I would like to use different strategies for these errors.